source/unified-test-format/unified-test-format.rst

```diff
@@ -1798,6 +1846,159 @@ Use a ``listIndexes`` command to check whether the index exists. Note that it is
+considered an error.
+
+This specification permits the test runner to report some failures as errors
+and some errors as failures. When the test runner stores errors and
+failures as entities it MAY classify conditions as errors and failures in
+the same way as it would when used in the driver's test suite.
+This includes reporting all errors as failures or all failures as errors.
+
+If the test runner does not distinguish errors and failures in its reporting,
+it MAY report both conditions under either category, but it MUST report
+any given condition in at most one category.
+
+The same entity name MAY be used for both ``storeErrorsAsEntity``
+and ``storeFailuresAsEntity`` options. The test runner MUST support
+writing both errors and failures to the same entity. If the same entity is
+specified, the errors and failures MUST be written in chronological order
+to the extent possible, interleaving the error and failure entries
+if necessary. Note that the textual description field for both errors
+and failures is ``error``.
+
+The following termination behavior MUST be implemented by the test
+runner:
+
+- The test runner MUST provide a way to request termination of loops. This
+  request will be made by the Atlas testing workload executor in response to
+  receiving the termination signal from Astrolabe.
+  
+- When the termination request is received, the test runner MUST
+  stop looping. If the test runner is looping when the termination request
+  is received, the current loop iteration MUST complete to its natural
+  conclusion (success or failure). If the test runner is not looping
+  when the termination request is received, it MUST NOT start any new
+  loop iterations in either the current test or subsequent tests for the
+  lifetime of the test runner.
+
+- The termination request MUST NOT affect non-loop operations, including
+  any operations after the loop. The tests SHOULD NOT be written in such
+  a way that the success or failure of operations that follow loops
+  depends on how many loop iterations were performed.
+  
+- Receiving the termination request MUST NOT by itself be considered an error
+  or a failure by the test runner.
+
+The exact mechanism by which the workload executor requests termination
+of the loop in the test runner, including the respective API, is left
+to the driver.
+
+An example of this operation follows::
+
+    - name: loop
+      object: testRunner
+      arguments:
+        storeErrorsAsEntity: errors
+        storeFailuresAsEntity: failures
+        storeSuccessesAsEntity: successes
+        storeIterationsAsEntity: iterations
+        operations:
+          - name: find
+            object: *collection0
+            arguments:
+              filter: { _id: { $gt: 1 }}
+              sort: { _id: 1 }
+            expectResult:
+              -
+                _id: 2
+                x: 22
+              -
+                _id: 3
+                x: 33
```

Comment:
nit: I think these can be condensed to a single line, like you did for `arguments`.

Reply:
I actually copy-pasted that from somewhere but I'll use a single line here as you are suggesting.